### PR TITLE
keyguard 1.6.0,20241002 (new cask)

### PR DESCRIPTION
**Pull request by nikhilbadyal**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/k/keyguard.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  livecheck do
+    url :url
+    regex(%r{/r(\d+)/Keyguard[._-](\d+(?:\.\d+)*)\.dmg}i)
+    strategy :github_latest do |json, regex|
+      json["assets"]&.map do |asset|
+        match = asset["browser_download_url"]&.match(regex)
+        next if match.blank?
+
+        "r#{match[1]},#{match[2]}"
+      end
```
Comment (khipp) [inline Casks/k/keyguard.rb:19]:
```suggestion
    regex(%r{/r?(\d+(?:\.\d+)*)/Keyguard[._-](\d+(?:\.\d+)+)\.dmg}i)
    strategy :github_latest do |json, regex|
      json["assets"]&.map do |asset|
        match = asset["browser_download_url"]&.match(regex)
        next if match.blank?

        "#{match[2]},#{match[1]}"
      end
```

This should make the regex more robust against changes in the tag naming scheme.

File: Casks/k/keyguard.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+cask "keyguard" do
+  version "r20241002,1.6.0"
+  sha256 "3631112e36eb33d5b9c7d2019bd418650c777d98b1fe2af1f9797e47364bf0ef"
+
+  url "https://github.com/AChep/keyguard-app/releases/download/#{version.csv.first}/Keyguard-#{version.csv.second}.dmg"
```
Comment (khipp) [inline Casks/k/keyguard.rb:5]:
```suggestion
  version "1.6.0,20241002"
  sha256 "3631112e36eb33d5b9c7d2019bd418650c777d98b1fe2af1f9797e47364bf0ef"

  url "https://github.com/AChep/keyguard-app/releases/download/r#{version.csv.first}/Keyguard-#{version.csv.second}.dmg"
```

We should list the app version first.

**Review by khipp — COMMENTED** [review]
Thank you for your contribution, @nikhilbadyal! I left a few minor suggestions below.

File: Casks/k/keyguard.rb (hunk 2 above)
Comment (nikhilbadyal) [reply Casks/k/keyguard.rb:5]: https://github.com/AChep/keyguard-app/releases/download/r20241002/Keyguard-1.6.0.dmg

But the URL pattern is like. Won't above changes break the URL ?

Comment (khipp) [reply Casks/k/keyguard.rb:5]: Thanks, that was an oversight on my part. I have updated the suggestion.

**Review by khipp — APPROVED** [review]
Thank you!

**Pull request merged**